### DISCUSSION TITLE
umi4 如何添加空白页loading

### DISCUSSION THREAD

**Discussion by Ggmid** [umijs/umi#8366] [Q&A]
在umi3中，我可以在documnet.ejs里面添加自定义loading，在umi4中如何添加

**Comment by fz6m** [comment]
See https://umijs.org/docs/guides/directory-structure#loadingtsxjsx

**Reply by Ggmid** [reply]
> See https://umijs.org/docs/guides/directory-structure#loadingtsxjsx

这个解决不了刚进来的白屏问题

**Reply by fz6m** [reply]
给一个复现 demo 看看

**Reply by Ggmid** [reply]
> 给一个复现 demo 看看

找到了，已解决谢谢

**Comment by pzc666** [comment]
这个问题怎么解决的？

**Reply by Ggmid** [reply]
我是这么解决的，在项目的plugin.ts 里面通过api.modifyHTML和api.addHTMLStyless两个api组合添加html结构
